app/sso.js

```diff
@@ -0,0 +1,45 @@
+const crypto = require('crypto');
+const fs = require('fs');
+
+function generateToken() {
+	// Normally we would use 16 => 128 bits of entropy which is sufficiennt
+	// But since we're going to base64 it and there would be padding (==),
+	// It's a wasted space, why not use padding for some additional entropy? :)
+	// The replacing is to make it URL-safe
+	return crypto.randomBytes(18).toString("base64").replace('+', '-').replace('/', '_')
```

Comment:
This will only replace the first occurrence of `+` or `/`. Should use `/replace(/\+/g, '-').replace(/\\/g, '_')` to replace them all.

Reply:
Damn, thanks for heads up!